src/librustc_metadata/diagnostics.rs

```diff
@@ -26,6 +26,27 @@ name. Example:
 ```
 "##,
 
+E0455: r##"
+A framework link kind was used and the OS used to compile isn't the OS X
+cause native frameworks are only available on OS X targets.
+Erroneous code example:
+
+```compile_fail"
+#[link(name = "FooCoreServices",  kind = "framework")] extern {}
+// OS used to compile is Linux for example
+```
+
+When using 'framework' link kind, please ensure to use OSX to compile the code.
+Or you can use the Conditional Compilation:
+
+```
+#[cfg(target_os="macos")]
```

Comment:
I've tried that approach but it seems to have a syntax error. Take a look: http://is.gd/XYidSe